File: api/src/main/java/com/ruby/rubia_server/core/entity/AudioMessage.java
package com.ruby.rubia_server.core.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "audio_messages")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class AudioMessage {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "message_id", unique = true, nullable = false)
    private String messageId;

    @Column(name = "from_number", nullable = false)
    private String fromNumber;

    @Column(name = "to_number")
    private String toNumber;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "direction", nullable = false)
    private MessageDirection direction;

    @Column(name = "audio_url")
    private String audioUrl;

    @Column(name = "file_path")
    private String filePath;

    @Column(name = "mime_type")
    private String mimeType;

    @Column(name = "duration_seconds")
    private Integer durationSeconds;

    @Column(name = "file_size_bytes")
    private Long fileSizeBytes;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    @Column(name = "status", nullable = false)
    @Builder.Default
    private ProcessingStatus status = ProcessingStatus.RECEIVED;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "processed_at")
    private LocalDateTime processedAt;

    @Column(name = "error_message")
    private String errorMessage;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conversation_id")
    private Conversation conversation;

    public enum MessageDirection {
        INCOMING, OUTGOING
    }

    public enum ProcessingStatus {
        RECEIVED, DOWNLOADING, PROCESSING, COMPLETED, FAILED
    }
}

File: api-dotnet/Rubia.Server/Entities/AudioMessage.cs
using Rubia.Server.Enums;

namespace Rubia.Server.Entities;

public class AudioMessage
{
    public Guid Id { get; set; }
    public string MessageId { get; set; } = string.Empty;
    public string FromNumber { get; set; } = string.Empty;
    public string? ToNumber { get; set; }
    public MessageDirection Direction { get; set; }
    public string? AudioUrl { get; set; }
    public string? FilePath { get; set; }
    public string? MimeType { get; set; }
    public int? DurationSeconds { get; set; }
    public long? FileSizeBytes { get; set; }
    public ProcessingStatus Status { get; set; } = ProcessingStatus.RECEIVED;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ProcessedAt { get; set; }
    public string? ErrorMessage { get; set; }
    public long Version { get; set; }
    public Guid? ConversationId { get; set; }
    public Conversation? Conversation { get; set; }
}

public enum MessageDirection
{
    INCOMING,
    OUTGOING
}

public enum ProcessingStatus
{
    RECEIVED,
    DOWNLOADING,
    PROCESSING,
    COMPLETED,
    FAILED
}